bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogMetadata.java

```diff
@@ -86,10 +74,20 @@ public long getRemainingSize() {
         return remainingSize;
     }
 
-    Map<Long, Long> getLedgersMap() {
+    ConcurrentLongLongHashMap getLedgersMap() {
         return ledgersMap;
     }
 
+    public void removeLedgerIf(LongPredicate predicate) {
+        ledgersMap.removeIf((ledgerId, size) -> {
+            boolean shouldRemove = predicate.test(ledgerId);
+            if (shouldRemove) {
+                remainingSize -= size;
```

Comment:
This is interesting, if we had a standard ConcurrentHashMap the count could be wrong.
But in our implementation the predicate is executed only once.